lib/utils/inBetween.ts

```diff
@@ -0,0 +1,249 @@
+import { XSD } from "@treecg/types";
+import { getLoggerFor } from "./logUtil";
+
+const logger = getLoggerFor("InBetweenRelation");
+export type Between = {
+    min: Date;
+    max: Date;
+};
+
+export function parseInBetweenRelation(
+    value: string,
+    dataType: string | undefined,
+    defaultTimezone: string,
+): Between | undefined {
+    let thisMin: Date | undefined;
+    let thisMax: Date | undefined;
+
+    const updateResult = (min: Date, max: Date) => {
+        if (thisMin === undefined || min <= thisMin) {
```

Comment:
If I understand your refactor correctly, `thisMin` and `thisMax` will in your code always be `undefined`. However, what I initially did, is check with the current `min` and `max` to check if the `eqMin` and `eqMax` should be adjusted, based on any earlier relations set.
Could it be that you don't check earlier set relations, or am I overlooking something?